src/LfMergeBridge/LanguageForgeGetChorusNotesActionHandler.cs

```diff
@@ -37,90 +38,125 @@ internal sealed class LanguageForgeGetChorusNotesActionHandler : IBridgeActionTy
+						Replies = new List<LfCommentReply>(ann.Messages.Skip(1).Where(m => !String.IsNullOrWhiteSpace(m.Text)).Select(ReplyFromChorusMsg)),
+						IsDeleted = false,
+						Regarding = new LfCommentRegarding {
+							TargetGuid = ExtractGuidFromChorusRef(ann.RefStillEscaped),
+							// Word and Meaning will be set in LfMerge, but set them to something vaguely sensible here as a fallback
+							Word = ann.LabelOfThingAnnotated,
+							Meaning = string.Empty
+						}
 					};
 					lfComments.Add(lfComment);
 				}
 			}
-			var serializedComments = new StringBuilder("New comments not yet in LF: ");
-			serializedComments.Append(JsonConvert.SerializeObject(lfComments));
-			LfMergeBridgeUtilities.AppendLineToSomethingForClient(ref somethingForClient, serializedComments.ToString());
 
-			var serializedReplies = new StringBuilder("New replies on comments already in LF: ");
-			serializedReplies.Append(JsonConvert.SerializeObject(lfReplies));
-			LfMergeBridgeUtilities.AppendLineToSomethingForClient(ref somethingForClient, serializedReplies.ToString());
+			if (hadExtraData)
+			{
+				// This version of LfMerge follows the ExtraData protocol, so pass the objects back directly
+				var response = new GetChorusNotesResponse {
+					LfComments = lfComments,
+					LfReplies = lfReplies,
+					LfStatusChanges = lfStatusChanges,
+				};
+				// LfMergeBridge.ExtraOutputData.AddOrUpdate(options, response); // Not available in netstandard2.0
+				LfMergeBridge.ExtraOutputData.Remove(options); // Available in netstandard2.0, does not throw if value does not exist
+				LfMergeBridge.ExtraOutputData.Add(options, response); // Now this is guaranteed safe (would have thrown if previous value had not been removed)
+			}
+			else
+			{
+				// Older version of LfMerge, serialize (likely slow)
+				var serializedComments = new StringBuilder("New comments not yet in LF: ");
+				serializedComments.Append(JsonConvert.SerializeObject(lfComments));
+				LfMergeBridgeUtilities.AppendLineToSomethingForClient(ref somethingForClient, serializedComments.ToString());
+
+				var serializedReplies = new StringBuilder("New replies on comments already in LF: ");
+				serializedReplies.Append(JsonConvert.SerializeObject(lfReplies));
+				LfMergeBridgeUtilities.AppendLineToSomethingForClient(ref somethingForClient, serializedReplies.ToString());
 
-			var serializedStatusChanges = new StringBuilder("New status changes on comments already in LF: ");
-			serializedStatusChanges.Append(JsonConvert.SerializeObject(lfStatusChanges));
-			LfMergeBridgeUtilities.AppendLineToSomethingForClient(ref somethingForClient, serializedStatusChanges.ToString());
+				var serializedStatusChanges = new StringBuilder("New status changes on comments already in LF: ");
+				serializedStatusChanges.Append(JsonConvert.SerializeObject(lfStatusChanges));
+				LfMergeBridgeUtilities.AppendLineToSomethingForClient(ref somethingForClient, serializedStatusChanges.ToString());
+			}
```

Comment:
I clarified this in person, and I don't think we want to keep the old implementation around.  We are switching to the new implementation that avoids string serialization of comments.  This is an opportunity to reduce complexity in our code.

Reply:
Done in commit b572aa50.